### ISSUE TITLE
The recommended way of marking substitutional disorder

### ISSUE THREAD

**Issue by vaitkus** [COMCIFS/cif_core#251]
One popular way of expressing substitutional disorder in CIF files relies on grouping atoms with partial occupancies based on their identical coordinates. For example, the following fragment:
```
loop_
_atom_site_label
_atom_site_type_symbol
_atom_site_fract_x
_atom_site_fract_y
_atom_site_fract_z
_atom_site_occupancy
Sr Sr  0.99275 0.25000 0.49747 0.5
La La  0.99275 0.25000 0.49747 0.5
Fe Fe  0.00000 0.00000 0.00000 1.0
O1 O  0.00150 0.25000 0.93695  0.8333
F1 F  0.00150 0.25000 0.93695  0.1667
O2 O  0.23763 0.03458 0.26096  0.8333
F2 F  0.23763 0.03458 0.26096  0.1667
```
Indicates that one site is simultaneously occupied by Sr/La, another by O1/F1 and the third one by O2/F2. This type of markup conveys the general idea, however, it feels somewhat imprecise due to the potential rounding errors that one has to take into account while comparing the coordinates, etc. Also I was not able to find official documentation that describes this approach (I would be happy to get a reference, though).

Alternative approach uses the `_atom_site.disorder_group` and `_atom_site.disorder_assembly` data items from the `CIF_CORE` dictionary to group the disordered atoms in a more formal machine-readable way. However, it not clear if this approach is legit, since the definitions of both of these items explicitly use the term "positional disorder" and do not mention "substitutional  disorder" at all. Previous example, rewritten using the `_atom_site.disorder_*` items:
```
loop_
_atom_site_label
_atom_site_type_symbol
_atom_site_fract_x
_atom_site_fract_y
_atom_site_fract_z
_atom_site_occupancy
_atom_site.disorder_group
_atom_site.disorder_assembly
Sr Sr  0.99275 0.25000 0.49747 0.5 1 A
La La  0.99275 0.25000 0.49747 0.5 2 A
Fe Fe  0.00000 0.00000 0.00000 1.0 . .
O1 O  0.00150 0.25000 0.93695  0.8333 1 B
F1 F  0.00150 0.25000 0.93695  0.1667 2 B
O2 O  0.23763 0.03458 0.26096  0.8333 1 C
F2 F  0.23763 0.03458 0.26096  0.1667 2 C
```
Are there any COMCIFS recommendations  on marking substitutional disorder? Can the `_atom_site.disorder_*` data items be used for this purpose?

**Comment by jamesrhester** [comment]
p103 and 104 of IT Vol G 1st Edition state that occupational disorder can either be described as above, or else a dummy atom type can be created and the `atom_type` loop used to assign weighted values for the relevant parameters. While not stated, I think the former approach retains more information and is to be preferred.
Since the 1st edition was published a "restraints and constraints" dictionary has been developed, which allows the restraints on atom occupancies and positions to be made explicit.  Use of those data names would allow checking that the positions in data files match when they correspond to occupational disorder.

**Comment by vaitkus** [comment]
Thank you for the useful reference!

Slight deviation regarding the terminology. It seems that what I called _substitutional disorder_ (based on [1]) is referred to as _compositional disorder_ in IT Vol. G 1st Edition so I will be using this term from now on. Also, _occupational disorder_ seems to be a synonym of _positional disorder_ at least from the way these terms are used in p105 and p212 (definition of `_atom_site_disorder_group`) so you probably meant "compositional disorder" instead of "occupation disorder" in your previous message. It might be useful to write these definitions up (including the synonyms) in a resource that is slightly easier to search through than IT Vol. G. (e.g. the Online Dictionary of Crystallography [2]), though, I am slightly hesitant to do this myself due to the lack of expertise.

Regardless, now it is completely clear that there are two formalised ways of describing compositional disorder (_same site_ approach and _dummy site_ approach). However, one question still remains -- can the `_atom_site.disorder_*` items be used in combination with the same site approach to describe compositional disorder or is their usage strictly restricted to the situations of occupational (positional) disorder?

[1] https://ocw.mit.edu/courses/chemistry/5-067-crystal-structure-refinement-fall-2009/lecture-notes/MIT5_067F09_lec4.pdf
[2] https://dictionary.iucr.org/Main_Page

**Comment by jamesrhester** [comment]
Interesting. I think that the authors of the Volume and definitions had in mind groups of bonded atoms when defining the `_atom_site_disorder_*` tags. However, as noted in the original issue, these tags would work perfectly well for describing compositional disorder and are easier for machines to work with. Unfortunately both of the other methods must also be supported anyway.  I'll check to see if the core DMG have any objections.

**Comment by jamesrhester** [comment]
The core DMG have no objections to compositional disorder being described using `_atom_site.disorder_*` tags. We should update the relevant definitions and I will take care of the relevant chapter of the next edition of Volume G. It would be useful if an example could be prepared of the use of these tags to describe compositional disorder.

**Comment by vaitkus** [comment]
Relevant core DMG discussion can be found at https://www.iucr.org/__data/iucr/lists/coredmg/msg00414.html.

**Comment by vaitkus** [comment]
The definitions were updated and a few example CIF files were added by merging PR #343.

I will leave to @jamesrhester to close this issue in case he still wants to keep it open as a reminder to also update the text in Volume G.

Also, I will reiterate my question from the last comment of PR #343. I noticed that the disorder example that was provided in the DDL1 version of the dictionary as part of the ATOM_SITE category description (see https://github.com/COMCIFS/DDL1-legacy-dictionaries/blob/28e20dc928790dceb889716d2bed435fc10c4c79/dictionaries/cif_core.dic#L850) is no longer available in the DDLm dictionary. Maybe this example should be restored or a similar one reintroduced?

**Comment by jamesrhester** [comment]
Yes, the missing example should be returned, it is very helpful.
